packages/vite/src/node/baseEnvironment.ts

```diff
@@ -20,6 +20,7 @@ export function getDefaultResolvedEnvironmentOptions(
     optimizeDeps: config.optimizeDeps,
     dev: config.dev,
     build: config.build,
+    plugins: config.plugins,
```

Comment:
For environments that aren't in the config, we are using here the config plugins without expanding `applyToEnvironment`. Maybe we should directly disallow the creation of an environment that isn't in the resolved config.

Reply:
I think disallowing makes sense. I guess `configureEnvironment` hook won't run on them and that would be confusing.